--- Cabal.orig/src/Distribution/Simple.hs	2022-07-27 22:14:44.000000000 +0530
+++ Cabal.mod/src/Distribution/Simple.hs	2023-01-23 16:09:35.832474259 +0530
@@ -109,8 +109,17 @@
 import Distribution.Compat.Environment      (getEnvironment)
 import Distribution.Compat.GetShortPathName (getShortPathName)
 
-import Data.List       (unionBy, (\\))
+import Data.List       (unionBy, (\\), take, drop, isInfixOf)
 
+correctHostTriplet :: String -> String
+correctHostTriplet s = do
+  if isInfixOf "-android" s
+    then
+      let arch = take (length s -8) ( drop 0 s ) -- drop "-android"
+      in
+        if arch == "arm" then "armv7a" else arch ++ "-linux-"
+          ++ if arch == "arm" then "androideabi" else "android"
+    else s
 
 -- | A simple implementation of @main@ for a Cabal setup script.
 -- It reads the package description file using IO, and performs the
